### PR TITLE
feat: reflect Postgres errors on Tablespaces in cluster status

### PR DESCRIPTION
**Pull request by jsilvela**

### CODE REVIEW

File: api/v1/cluster_types.go [hunk 1]
```diff
@@ -679,7 +686,7 @@ type ClusterStatus struct {
 
 	// TablespaceStatus reports the state of the declarative tablespaces in the cluster
 	// +optional
-	TablespaceStatus TablespacesState `json:"tablespacesStatus,omitempty"`
+	TablespaceStatus []TablespaceState `json:"tablespaceStatus,omitempty"`
```
Comment (jsilvela) [inline api/v1/cluster_types.go:689]: Going to rename this field to `TablespacesStatus`. We're already using `TablespaceSatus` as an enum pending/reconciled.

Comment (jsilvela) [reply api/v1/cluster_types.go:689]: Don't love it but I think it helps clarify

**Review by jsilvela — COMMENTED** [review]
This is a beautiful PR @leonardoce . Thank you.

**Review by jsilvela — COMMENTED** [review]
Just made some corrections. BTW I am unable to generate the apidoc in my machine.
I give this a thumbs-up but cannot approve it as I created the PR

File: internal/management/controller/tablespaces/actions.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+	contextLog.Trace("creating tablespace ", "tablespace", r.tablespace.Name)
+	if exists, err := tbsStorageManager.storageExists(r.tablespace.Name); err != nil || !exists {
+		contextLog.Debug("deferring tablespace until creation of the mount point for the new volume",
+			"tablespaceName", r.tablespace.Name,
+			"tablespacePath", tbsStorageManager.getStorageLocation(r.tablespace.Name))
+		return apiv1.TablespaceState{
+			Name:  r.tablespace.Name,
+			State: apiv1.TablespaceStatusPendingReconciliation,
+			Error: "deferred until mount point is created",
```
[the remark this answers is not in the document]
Comment (litaocdl) [reply internal/management/controller/tablespaces/actions.go:53]: I agree, it is great, we can do similar for role

**Review by litaocdl — APPROVED** [review]

**Pull request merged**